app/packs/src/apps/mydb/elements/details/reactions/variationsTab/ReactionVariationsAnalyses.js

```diff
@@ -0,0 +1,184 @@
+import React, { useState } from 'react';
+import { AgGridReact } from 'ag-grid-react';
+import PropTypes from 'prop-types';
+import {
+  Label, FormGroup, Checkbox, Button, Modal
+} from 'react-bootstrap';
+import cloneDeep from 'lodash/cloneDeep';
+import Reaction from 'src/models/Reaction';
+import UIActions from 'src/stores/alt/actions/UIActions';
+import { getVariationsRowName } from 'src/apps/mydb/elements/details/reactions/variationsTab/ReactionVariationsUtils';
+
+function getReactionAnalyses(reaction) {
+  const analysesContainer = reaction.container?.children?.find((child) => child.container_type === 'analyses');
+  const analyses = analysesContainer?.children?.filter((analysis) => !analysis.is_new);
+
+  return analyses ?? [];
+}
+
+function updateAnalyses(variations, allReactionAnalyses) {
+  const analysesIDs = allReactionAnalyses.filter((analysis) => !analysis.is_deleted).map((child) => child.id);
+  const updatedVariations = cloneDeep(variations);
+  updatedVariations.forEach((row) => {
+    row.analyses = row.analyses.filter((id) => analysesIDs.includes(id));
+  });
+
+  return updatedVariations;
+}
+
+function getAnalysesOverlay({ data: variationsRow, context }) {
+  const { analyses: analysesIDs } = variationsRow;
+  const { allReactionAnalyses } = context;
+
+  return allReactionAnalyses.filter((analysis) => analysesIDs.includes(analysis.id));
+}
+
+function AnalysisOverlay({ value: analyses }) {
+  if (analyses.length === 0) {
+    return ''; // Don't return null, it breaks AG's logic to determine if component is rendered.
+  }
+  return (
+    <div
+      className="custom-tooltip"
+      style={{
+        padding: '3px 8px',
+        color: '#fff',
+        backgroundColor: '#000',
+        borderRadius: '4px',
+      }}
+    >
+      Linked analyses:
+      <ul style={{ paddingLeft: '15px' }}>
+        {analyses.map((analysis) => (
+          <li key={analysis.id}>{analysis.name}</li>
+        ))}
+      </ul>
+
+    </div>
+  );
+}
+
+AnalysisOverlay.propTypes = {
+  value: PropTypes.instanceOf(AgGridReact.value).isRequired,
+};
+
+function AnalysisVariationLink({ reaction, analysisID }) {
+  const { variations } = cloneDeep(reaction);
+  const linkedVariations = variations.filter((row) => row.analyses.includes(analysisID)) ?? [];
+
+  if (linkedVariations.length > 0) {
```

Comment:
nit: inconsistent: in function `AnalysisOverlay` error case is checked with early return, in `AnalysisVariantLink` its checked for the normal case and returned.

Reply:
Nice, an eye for consistency :monocle_face: Fixed with 515972d2f4efb6a89c526c8cbc01bcb4744269ed.